src/db.rs

```diff
@@ -26,6 +31,29 @@ use crate::transactions::SAVEPOINT_TABLE;
 #[cfg(feature = "logging")]
 use log::{info, warn};
 
+/// Implements persistent storage for a database.
+pub trait StorageBackend: 'static + Debug + Send + Sync {
+    /// Whether the current storage backend is of zero length.
+    fn is_empty(&self) -> Result<bool, io::Error> {
+        Ok(self.len()? == 0)
+    }
```

Comment:
I don't think this is used. Let's remove it?

Reply:
Indeed, it is not used - I added this because of the Clippy lint that forbids having a `len` method without `is_empty`. I can remove it and add an `#[allow(clippy::len_without_is_empty)]` if that would be better!

Reply:
Oh ya, let's add that `allow`. That seems like a silly Clippy rule :)